### ISSUE TITLE
Expose default random number generator for XLA devices

### ISSUE THREAD

**Issue by lqf96** [pytorch/xla#2532]
## 🚀 Feature
It would be nice to expose default random generator for XLA devices, just like CPU (at `torch.default_generator`) and GPU (at `torch.cuda.default_generators`) devices in PyTorch. The default generators can live in e.g. `torch_xla.core.random.default_generators` and the `torch_xla.core.random` can additionally provide a number of methods to get, set and seed the random number generator states (e.g. `seed`, `manual_seed`, `get_rng_state`, `set_rng_state`).

## Motivation
This will be helpful for reproducing the results of neural networks when using TPUs.

**Comment by JackCaoG** [comment]
You can set the seed via 
https://github.com/pytorch/xla/blob/df0a2776e42ab144cfb00485c50852ea70f2b9b8/torch_xla/core/xla_model.py#L917

some example in 
https://github.com/pytorch/xla/blob/132481bbfd1efa7a06e40ba725a85c86023828c3/test/test_operations.py#L81

Will this be enough for your use case?

**Comment by lqf96** [comment]
@JackCaoG Yeah I know that... But I have some code that operates on `torch.Generator`s, and it would be more convenient if `default_generators` are provided and `torch.random.manual_seed` extends support to XLA devices.

**Comment by JackCaoG** [comment]
Pytorch/XLA runs on XLA device, we used [RngBitGenerator](https://www.tensorflow.org/xla/operation_semantics#rngbitgenerator) OP to generate the random number (where we can only select an algorithm and set the initial state/seed). We don't use the `torch.Generator` and I can't think of a way to fit this into the Pytorch/XLA.

Even if we can call generator on CPU and then feed the constant to the graph, the constant will change every step and we will keep recompiling the graph.

**Comment by lqf96** [comment]
Ok I read some PyTorch and XLA source code and noticed that this will be, in principle, doable. In PyTorch each `torch.Generator` is a [`aten::Generator`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/Generator.h) in C++, which in turn is implemented by a device-specific subclass of [`c10::GeneratorImpl`](https://github.com/pytorch/pytorch/blob/master/c10/core/GeneratorImpl.cpp). Also notice that in XLA, all random operations return both results and new random states, as shown by the [`RngOutput`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/client/lib/prng.h) struct, but as of now PyTorch XLA ignores the random states. So one idea to implement this will be:

1. Make a subclass of `c10::GeneratorImpl` for XLA devices, possibly called `XLAGeneratorImpl` and initialize this generator with a XLA tensor as the RNG state which internally contains a XLA graph node.
2. When performing random-related operations, use the current `XLAGeneratorImpl` state as the input RNG state node and then replace the `XLAGeneratorImpl` state with the output RNG state node.
3. Under this design the XLA Generator will also be lazy and the actual random state is only revealed (or executed) when APIs like `aten::Generator::current_seed` or `Generator.get_state` is called.

I can look into this and try to come up with a PR if I have time.
